### PR TITLE
fix cpydiff generation on platforms where 'python' executable is Python 2

### PR DESCRIPTION
**Pull request by blazewicz**
This patch should also fix building docs on windows.

### CODE REVIEW

**Comment by pfalcon** [comment]
I surely hit this issue too, +1.

File: docs/Makefile [hunk 1]
```diff
@@ -60,7 +61,7 @@ clean:
 cpydiff:
 	@echo "Generating MicroPython Differences."
 	rm -f $(GENRSTDIR)/*
-	cd $(CPYDIFFDIR) && python $(CPYDIFF)
+	cd $(CPYDIFFDIR) && $(PYTHON) $(CPYDIFF)
```
Comment (stinos) [inline docs/Makefile]: Just wondering, would it also not be possible to use the environment variable MICROPY_CPYTHON3 here?

**Comment by blazewicz** [comment]
It certainly would be possible, and I could have written:

```
MICROPY_CPYTHON3 ?= python3
```

But I think that purpose of those two variables is slightly different.

`PYTHON` used in Makefiles is an interpeter compatible with tools used in build process, whereas `MICROPY_CPYTHON3` is an executable used to test uPy against.

```
$ grep -R '$(PYTHON)' .
./esp8266/Makefile:	$(Q)$(PYTHON) makeimg.py $(BUILD)/firmware.elf-0x00000.bin $(BUILD)/firmware.elf-0x[0-5][1-f]000.bin $@
./esp8266/Makefile:#	$(Q)$(PYTHON) $(MAKE_PINS) --board $(BOARD_PINS) --af $(AF_FILE) --prefix $(PREFIX_FILE) --hdr $(GEN_PINS_HDR) --qstr $(GEN_PINS_QSTR) --af-const $(GEN_PINS_AF_CONST) --af-py $(GEN_PINS_AF_PY) > $(GEN_PINS_SRC)
./teensy/Makefile:	$(Q)$(PYTHON) $(MAKE_MEMZIP) --zip-file $(BUILD)/memzip-files.zip --c-file $@ $(MEMZIP_DIR)
./teensy/Makefile:	$(Q)$(PYTHON) $(MAKE_PINS) --board $(BOARD_PINS) --af $(AF_FILE) --prefix $(PREFIX_FILE) --hdr $(GEN_PINS_HDR) --qstr $(GEN_PINS_QSTR) --af-const $(GEN_PINS_AF_CONST) --af-py $(GEN_PINS_AF_PY) > $(GEN_PINS_SRC)
./minimal/Makefile:	$(Q)$(PYTHON) $(DFU) -b 0x08000000:$(BUILD)/firmware.bin $@
./minimal/Makefile:	$(Q)$(PYTHON) $(PYDFU) -u $<
./lib/memzip/README.md:    $(Q)$(PYTHON) $(MAKE_MEMZIP) --zip-file $(BUILD)/memzip-files.zip --c-file $@ $(MEMZIP_DIR)
./docs/Makefile:	cd $(CPYDIFFDIR) && $(PYTHON) $(CPYDIFF)
./cc3200/application.mk:	$(Q)$(PYTHON) $(UPDATE_WIPY) --verify --ip $(WIPY_IP) --user $(WIPY_USER) --password $(WIPY_PWD) --file $<
./cc3200/application.mk:	$(Q)$(PYTHON) $(MAKE_PINS) --board $(BOARD_PINS) --af $(AF_FILE) --prefix $(PREFIX_FILE) --hdr $(GEN_PINS_HDR) --qstr $(GEN_PINS_QSTR) > $(GEN_PINS_SRC)
./py/py.mk:	$(Q)$(PYTHON) $(PY_SRC)/makeversionhdr.py $@
./py/py.mk:	$(Q)$(PYTHON) $(PY_SRC)/makeqstrdata.py $(HEADER_BUILD)/qstrdefs.preprocessed.h > $@
./py/mkrules.mk:	$(Q)$(PYTHON) $(PY_SRC)/makeqstrdefs.py split $(HEADER_BUILD)/qstr.i.last $(HEADER_BUILD)/qstr $(QSTR_DEFS_COLLECTED)
./py/mkrules.mk:	$(Q)$(PYTHON) $(PY_SRC)/makeqstrdefs.py cat $(HEADER_BUILD)/qstr.i.last $(HEADER_BUILD)/qstr $(QSTR_DEFS_COLLECTED)
./py/mkrules.mk:	$(Q)$(PYTHON) $(MPY_TOOL) -f -q $(BUILD)/genhdr/qstrdefs.preprocessed.h $(FROZEN_MPY_MPY_FILES) > $@
./stmhal/Makefile:	$(Q)$(PYTHON) $(PYDFU) -u $<
./stmhal/Makefile:	$(Q)$(PYTHON) $(DFU) -b $(FLASH_ADDR):$(BUILD)/firmware0.bin -b $(TEXT_ADDR):$(BUILD)/firmware1.bin $@
./stmhal/Makefile:	$(Q)$(PYTHON) $(MAKE_PINS) --board $(BOARD_PINS) --af $(AF_FILE) --prefix $(PREFIX_FILE) --hdr $(GEN_PINS_HDR) --qstr $(GEN_PINS_QSTR) --af-const $(GEN_PINS_AF_CONST) --af-py $(GEN_PINS_AF_PY) > $(GEN_PINS_SRC)
./stmhal/Makefile:	$(Q)$(PYTHON) make-stmconst.py --qstr $(GEN_STMCONST_QSTR) --mpz $(GEN_STMCONST_MPZ) $(CMSIS_MCU_HDR) > $(GEN_STMCONST_HDR)
./stmhal/Makefile:	$(Q)$(PYTHON) $(FILE2H) $< > $@
./stmhal/Makefile:	$(Q)$(PYTHON) $(INSERT_USB_IDS) $(USB_IDS_FILE) $< > $@
```

**Comment by blazewicz** [comment]
I took `MICROPY_CPYTHON3` from `tests/run-tests`.

**Comment by stinos** [comment]
Good point, forgot check how it's used in the other makefiles myself.

**Comment by dpgeorge** [comment]
Thanks!  Merged in ae116c2430f9c990dcf619f4b8f90b2fc0c18984 and 23ccb3e12e3b7fe6b56785d1e17e9d990d1da86c

**Pull request closed without merging**